src/schema/posts.ts

```diff
@@ -1101,6 +1223,121 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
   BaseContext
 >({
   Query: {
+    sourcePostModeration: async (
+      _,
+      { id, sourceId }: { id: string; sourceId: string },
+      ctx: Context,
+      info,
+    ): Promise<GQLSourcePostModeration> => {
+      await ensureSourcePermissions(
+        ctx,
+        sourceId,
+        SourcePermissions.ModeratePost,
+      );
```

Comment:
I think that if we are going to update this line we should update even the `ensureSourcePermissions` function and put an additional if `permission === SourcePermissions.ModeratePost` check (if user is mod or is the current author). 

Plus: maybe `ModeratePost` is not the best name for this. What do you think about something like `ViewModerationPost`?

Reply:
I think that's going to be very clunky because it's going to require us to pass the Moderation Entity into ensureSourcePermissions as well, or fetch it first and then send the author id.

I think in this case it would then be better to again check isPrivilegedMember, and if not, try add an additional "andWhere userId" clause on the graphorm query.

Thoughts @capJavert  ?

Reply:
I'm open for both approaches ✔️ 

My main concern is that we added a value to enum that is not reflecting action (view post) nor what the method is doing.

Reply:
Do agree with Luca that enum value does not match.

Since this is just read operation again, it looks more simple to me to use `isPrivilegedMember` in this case indeed. 

Let's focus on what needs to be done:
- user should not be able to get post moderation when he is not a member of source
- user should be able to get its post moderation
- admin/moderator should be able to get all post moderations for source

Anything else?

If above is implemented and covered with tests to me its good.